pandas/core/reshape/merge.py

```diff
@@ -1200,7 +1209,34 @@ def _maybe_coerce_merge_keys(self):
                 typ = rk.categories.dtype if rk_is_cat else object
                 self.right = self.right.assign(**{name: self.right[name].astype(typ)})
 
+    def _create_cross_configuration(
+        self, _left, _right, on
+    ) -> Tuple["DataFrame", "DataFrame", str, str]:
+        if on is not None:
+            raise MergeError(
+                "Can not pass any merge columns when using cross as merge method"
+            )
+        cross_col = f"{max([*_left.columns, *_right.columns])}_cross"
+        _left = _left.copy()
+        _right = _right.copy()
+        _left.insert(loc=0, value=1, column=cross_col)
+        _right.insert(loc=0, value=1, column=cross_col)
+        how = "inner"
+        on = cross_col
+        self._cross = cross_col
+        return _left, _right, how, on
+
     def _validate_specification(self):
+        if hasattr(self, "_cross"):
```

Comment:
instead of checking the attribute, rather can you check if how is 'cross'?

Reply:
Depending on answer above, this is done